pandas/core/reshape/merge.py

```diff
@@ -2067,17 +2063,26 @@ def _validate_operand(obj: FrameOrSeries) -> "DataFrame":
         )
 
 
-def _items_overlap_with_suffix(left: Index, lsuffix, right: Index, rsuffix):
+def _items_overlap_with_suffix(left: Index, right: Index, suffixes):
     """
+    Suffixes type validatoin.
+
     If two indices overlap, add suffixes to overlapping entries.
 
     If corresponding suffix is empty, the entry is simply converted to string.
 
     """
+    if not isinstance(suffixes, (tuple, list)):
+        raise TypeError(
+            f"suffixes should be of type list/tuple. But got {type(suffixes)}"
```

Comment:
also check that the len is exactly 2

Reply:
I have validated for type `tuple`. Should i do explicit validation of `tuple(str, str)`??